test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java

```diff
@@ -113,15 +112,13 @@ public static void assertConsistency(BlobStoreRepository repository, Executor ex
                 assertIndexUUIDs(repository, repositoryData);
                 assertSnapshotUUIDs(repository, repositoryData);
                 assertShardIndexGenerations(blobContainer, repositoryData.shardGenerations());
-                return null;
             } catch (AssertionError e) {
-                return e;
+                listener.onResponse(e);
+                return;
```

Comment:
nit: we could move `listener.onResponse(null)` at the end of the try block so we don't need this return. But this is just a suggestion 👍